docs/cog_guides/downloader.rst

```diff
@@ -0,0 +1,547 @@
+
+"""""""""""""""""""
+cog checkforupdates
+"""""""""""""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog checkforupdates 
+
+**Description**
+
+Check for available cog updates (including pinned cogs).
+
+This command doesn't update cogs, it only checks for updates.
+Use ``[p]cog update`` to update cogs.
+
+.. _downloader-command-cog-info:
+
+""""""""
+cog info
+""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog info <repo_name> <cog_name>
+
+**Description**
+
+List information about a single cog.
+
+Example:
+    - ``[p]cog info 26-Cogs defender``
+
+**Arguments**
+
+- ``<repo_name>`` The repo to get cog info from..
+- ``<cog_name>`` The cog to get info on.
+
+.. _downloader-command-cog-install:
+
+"""""""""""
+cog install
+"""""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog install <repo_name> <cogs>
+
+**Description**
+
+Install a cog from the given repo.
+
+Examples:
+    - ``[p]cog install 26-Cogs defender``
+    - ``[p]cog install Laggrons-Dumb-Cogs say roleinvite``
+
+**Arguments**
+
+- ``<repo_name>`` The name of the repo to install cogs from.
+- ``<cogs>`` The cog or cogs to install.
+
+.. _downloader-command-cog-installversion:
+
+""""""""""""""""""
+cog installversion
+""""""""""""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog installversion <repo_name> <revision> <cogs>
+
+**Description**
+
+Install a cog from the specified revision of given repo.
+
+Example:
+    - ``[p]cog installversion Broken-Repo e798cc268e199612b1316a3d1f193da0770c7016 cog_name``
+
+**Arguments**
+
+- ``<repo_name>`` The name of the repo to install cogs from.
+- ``<revision>`` The revision to install from.
+- ``<cogs>`` The cog or cogs to install.
+
+.. _downloader-command-cog-list:
+
+""""""""
+cog list
+""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog list <repo_name>
+
+**Description**
+
+List all available cogs from a single repo.
+
+Example:
+    - ``[p]cog list 26-Cogs``
+
+**Arguments**
+
+- ``<repo_name>`` The repo to list cogs from.
+
+.. _downloader-command-cog-listpinned:
+
+""""""""""""""
+cog listpinned
+""""""""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog listpinned 
+
+**Description**
+
+List currently pinned cogs.
+
+.. _downloader-command-cog-pin:
+
+"""""""
+cog pin
+"""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog pin <cogs>
+
+**Description**
+
+Pin cogs - this will lock cogs on their current version.
+
+Examples:
+    - ``[p]cog pin defender``
+    - ``[p]cog pin outdated_cog1 outdated_cog2``
+
+**Arguments**
+
+- ``<cogs>`` The cog or cogs to pin. Must already be installed.
+
+.. _downloader-command-cog-reinstallreqs:
+
+"""""""""""""""""
+cog reinstallreqs
+"""""""""""""""""
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cog reinstallreqs 
+
+**Description**
+
+This command will reinstall cog requirements and shared libraries for all installed cogs.
+
+Red might ask user to use this when it clears contents of lib folder
```

Comment:
```suggestion
Red might ask the owner to use this when it clears contents of thr lib folder
```

Reply:
While I do think this grammar change should be done in the docstring, this command should **not** be included in the docs  as it is marked as hidden.

Reply:
The command is not actually hidden, I've mistakenly told Flame that it is.
Nonetheless, I still don't think the command should really be shown in the user guide as it's generally only supposed to be used if Red asks you to and it's not a command that the user really needs to learn about from this guide.

Reply:
Also added a comment that this command shouldn't be used unless Red asks for it.

Reply:
Seems fine to me to hide it.

Reply:
Added PR to hide this in #4590